cellengine/resources/experiment.py

```diff
@@ -209,28 +210,46 @@ def undelete(self):
             del self.deleted
 
     @property
-    def active_compensation(self) -> Optional[Union[Compensation, int]]:
+    def active_compensation(self) -> Optional[Union[str, int]]:
+        """The most recently used compensation.
+
+        Accepted (case-insensitive) values are one of:
+            - A Compensation object,
+            - A Compensation ID,
+            - A `cellengine` module variable, one of: [UNCOMPENSATED,
+                FILE_INTERNAL, PER_FILE]
+            - One of [0, -1, -2]
+
+        Example:
+            ```python
+            import cellengine
+            exp = cellengine.get_experiment(name='my experiment')
+            experiment.active_compensation = cellengine.UNCOMPENSATED
+            ```
+        """
         active_comp = self._active_comp
-        if type(active_comp) is str:
-            return ce.APIClient().get_compensation(self._id, active_comp)
-        elif type(active_comp) is int:
+        if isinstance(active_comp, (str, int)):
             return active_comp
         elif active_comp is None:
             return None
-        raise ValueError(
-            f"Value '{active_comp}' is not a valid for activeCompensation."
-        )
+        else:
+            raise ValueError(
+                f"Value '{active_comp}' is not valid for active_compensation."
+            )
```

Comment:
With the new logic in the setter, you should just be able to `return self._active_comp` in this function.

Reply:
sure can; thanks!